### PR TITLE
uninstall flow delete all noobaa resources

### PR DESCRIPTION
**Pull request by naveenpaul1**
PR will have:

uninstall flow will delete all the Noobaa resources before deleting the namespace
Noobaa default backing store size change
automatic cache BucakeClass reaction code removes
Noobaa CR will have defaultbackingstore name for Front end team
Prometheus version update
Signed-off-by: Naveen Paul @naveenpaul1

### CODE REVIEW

File: hack/deploy/prometheus.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: downstream-prometheus-operator
+  namespace: redhat-data-federation
+spec:
+  displayName: downstream-prometheus-operator
+  image: quay.io/dbindra/ose-prometheus-operator:4.10.0_catsrc
```
Comment (rexagod) [inline hack/deploy/prometheus.yaml:8]: Can you please make sure that on running the following command,
```
curl -H "Authorization: Bearer `cat /var/run/secrets/kubernetes.io/serviceaccount/token`" --cacert /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt https://prometheus.redhat-data-federation.svc:9339/metrics
```
inside the `prometheus-k8s-0` pod (in `prometheus` container) in the `openshift-monitoring` namespace, the output is similar to the one below?
image
***
Can you attach a screenshot of the output (that you are getting) as well? Thanks.

File: controllers/mcg_bucket.go [hunk 2]
```diff
@@ -59,80 +59,19 @@ func (r *ManagedMCGReconciler) watchBucketClass(object client.Object) {
...
 			return
 		}
-		annotations := object.GetAnnotations()
-		if isCacheEnabled, ok := annotations[McgmsCacheEnabled]; ok && isCacheEnabled == "true" {
-			r.reconcileCacheBucketClass(object)
-		}
 	}
 }
 
-func (r *ManagedMCGReconciler) reconcileCacheBucketClass(object client.Object) {
```
Comment (rexagod) [inline controllers/mcg_bucket.go:69]: We don't need this anymore?

**Review by rexagod — CHANGES_REQUESTED** [review]
No changes, just a couple of comments.

File: controllers/mcg_bucket.go (hunk 2 above)
Comment (rexagod) [reply controllers/mcg_bucket.go:69]: > automatic cache BucketClass reaction code removes

Can you elaborate on this a bit? We don't need cache buckets now?

File: hack/deploy/prometheus.yaml (hunk 1 above)
Comment (naveenpaul1) [reply hack/deploy/prometheus.yaml:8]: @rexagod : yes or put is similar to this
image

Comment (rexagod) [reply hack/deploy/prometheus.yaml:8]: Awesome, LGTM!

File: controllers/mcg_bucket.go (hunk 2 above)
Comment (naveenpaul1) [reply controllers/mcg_bucket.go:69]: @rexagod before frond-end was passing the `single` type BucketClass for cached Bucket and we were creating cached BuketClass  along with OBC and single BucketClass had no purpose, but now if `cache` is selected in frond-end, front-end will create cache type BucketClass and backend only create OBC

Comment (rexagod) [reply controllers/mcg_bucket.go:69]: Oh okay, that's sounds much better. Thank you for clarifying this.

**Review by rexagod — APPROVED** [review]

**Pull request merged**